src/Controller/Api/TreeController.php

```diff
@@ -115,7 +116,7 @@ public function __construct(EntityManagerInterface $doctrineOrmEntityManager, En
      *
      * @return string
      */
-    public function getFundingOrganizationsAction(Request $request, DatasetIndex $datasetIndex)
+    public function getFundingOrganizationsAction(Request $request, DatasetIndex $datasetIndex, FundingOrgFilter $fundingOrgFilter)
```

Comment:
Add FundingOrgFilter to @param  above.

Reply:
FundingOrgFilter could be moved to the constructor as it is being used at more than one place

Reply:
I think that's kinda the beauty of local dependency injection. You only get an instance where it's needed, not for the entire controller. The constructor is how we HAD to do it  before this.

Reply:
Are you talking about before the FLEX upgrade? I think we used to get it from the container(). After flex upgrade, if a dependency injection is required for a Controller method, we use the local dependency injection or else if it is required more than once, we move it to the constructor. Just like entityManager or entityHandler